### ISSUE TITLE
Stop building both "legacy" and "non-legacy" Yarn versions

### ISSUE THREAD

**Issue by Daniel15** [yarnpkg/yarn#4082]
It's confusing having both "yarn" and "yarn-legacy". The only difference is that the "legacy" build is compiled for Node 4.0+, whereas the "modern" build is compiled for Node 6.0+ (so fewer transformations are ran). We're already only including the legacy version in the tarball, Debian package, and Windows installer, so the 'modern' one is essentially just being ignored. We should stop building it.

**Comment by BYK** [comment]
I tend to disagree here without getting some download metrics. I personally chose the modern build at my old company and I have a psychological tendency to believe that the "modern" version will be faster.

**Comment by moberhuber** [comment]
As a yarn newbie, I confirm having 2 variants is confusing - see also https://github.com/yarnpkg/website/issues/709

Perhaps rename them to just "yarn" for the version that runs everywhere, and "yarn-node6+" for the version that requires node6 or newer? - Because the term "legacy" seems to indicate lack of functionality which I don't think is true?
